### ISSUE TITLE
Need separable config for different architectures for nsis

### ISSUE THREAD

**Issue by mastergberry** [electron-userland/electron-builder#1306]
* **Version**: 14.4.0

* **Target**: nsis 32/64 bit

So we are starting to work towards a 32 bit build (right now supporting only 64 bit build).

Basically the entire `build {}` config thing does not really work for us going forward because we need to have different `files/extraFiles` for different builds. In theory the `publish` url's might change as well, the `nsis` include etc.

The proposed solution/migration is to make `build` into something like this:

```
"build": {
  "x64": {
    "appId": "...",
    "files": ......
  },
  "ia32": {
     ...
  }
}
```

By default if you do not have specific versions such as `x64` or `ia32` defined it will just default to the existing layout. If this is not too easy then maybe a `default` option for the `build` section.

**Comment by develar** [comment]
Why not just use macro? arch, os?

**Comment by develar** [comment]
> You can use ${os} (expanded to mac, linux or win according to current platform) and ${arch} in the pattern.

**Comment by mastergberry** [comment]
So you are proposing something like

```
"extraFiles": [
      "lib-${arch}/*.jar",
      "lib-${arch}/Proguard/**/*",
      {
        "from": "lib-${arch}/",
        "to": ".",
        "filter": [
          "*.dll"
        ]
      },
      {
        "from": "lib-${arch}/",
        "to": ".",
        "filter": [
          "*.exe"
        ]
      },
      "licenses/**",
      "BLClient/BLClient.json",
      {
        "from": "include/",
        "to": "."
      }
    ],
```

correct? if so I think this can be made to work also. I didn't really think about those being options.

**Comment by develar** [comment]
Yes.

**Comment by mastergberry** [comment]
Ok then this should suit my needs. I assume this can be done for anything including something like

`publish: [{"url": "https://example.com/${arch}/"}]` 

right?

if so this issue can be closed as my lack of understanding the documentation :) Thanks

**Comment by develar** [comment]
Yes, macros supported in the publish urls as well. But not for nsis.

**Comment by mastergberry** [comment]
So I'm reviewing the documentation quickly for `nsis-web` because I would like to switch to this since you made it for me to use in another issue xD

So i changed my build to use `nsis-web` and it outputted 3 files, the web installer exe, and two .7z files. There is also a `latest.yml` inside of `nsis-web/`. It only seems to have the x64 version inside of it though...also the github artifact name is completely wrong (not using this though). How do you make this work for both architectures?

**Comment by mastergberry** [comment]
Example:

```version: 0.9.3
releaseDate: '2017-02-26T09:45:21.029Z'
githubArtifactName: badlion-client-Setup-0.9.3.exe
path: badlion-client-0.9.3-x64.nsis.7z
sha2: b6bf1760b505aa5abf2832af8653de667664503cb0e640d7d76b9dfe14da4349```

notice the `githubArtifactName` is wrong and there is nothing in here for `ia32`

**Comment by develar** [comment]
> notice the githubArtifactName is wrong and there is nothing in here for ia32

@mastergberry Fixed in the 14.4.2
